astro.config.mjs

```diff
@@ -18,13 +18,13 @@ export default defineConfig({
   base,
   server: { port: config.port || 3000 },
 
-  outDir: join(coreDirectory, 'dist'),
+  outDir: join(coreDirectory, config.outDir || 'dist'),
```

Comment:
This `outDir` refers to `.eventcatalog-core`. Is this `dir` you want to change or [this](https://github.com/event-catalog/eventcatalog/blob/main/bin/eventcatalog.ts#L114)?

Reply:
Im not sure if that works. If I revert my changes  and then change the line you linked to `copyFolder(join(core, 'dist'), join(dir, 'dist-wahoo'));`, then run `npm run build`, and it still outputs it to `dist`. 

But if I change it in this astro config, it does actually build it into the correct folder

Reply:
To test this you need to do something similar to [this docker image](https://github.com/event-catalog/eventcatalog/blob/main/Dockerfile), that have been placed to catch bugs of this package on production (user's directory).

What I do when I want to test here:
1. run `npm run build:bin` - _builds the `eventcatalog`_
2. run `npm pack` - _generates the package that will be publish at npm registry_
3. `cd examples/default/` - _goes to the example emulating the user's directory_
4. `npm install ../../*.tgz` - _install the package generated on step 2_
5. `npx eventcatalog build` or `npx . build` - _runs the build command_

With this steps we can check if the `dist-wahoo` we be in `examples/default/`.

Reply:
Oh I see! Thank you for taking the time to walk me through that. I will test and make some changes later tonight

Reply:
Okay I got it working, thanks again. I definitely understand this codebase more after looking at those.

Though now Im getting a bit blocked on how to actually access that `outDir` value from within the npx build action. Do you have any guidance there?

If we arent able to access the config (or its a pain for some reason), perhaps we could modify the astro config still, and output to something like `astro-dist/${config.outDir}`

If we clear `astro-dist` before each build, then the single folder in there should be our target. Which we can just copy two levels up, no renaming necessary.

Fairly uninformed thoughts here, but hey :grin:  What do you think?

Reply:
> Though now Im getting a bit blocked on how to actually access that `outDir` value from within the npx build action. Do you have any guidance there?

You can access the `eventcatalog.config` using [this](https://github.com/event-catalog/eventcatalog/blob/main/scripts/eventcatalog-config-file-utils.js#L15)



> If we clear `astro-dist` before each build, then the single folder in there should be our target. Which we can just copy two levels up, no renaming necessary.

This is a fair point.

Another random thought about it. If we do something like below:
![image](https://github.com/user-attachments/assets/c2df6076-1bb8-4c0d-b72a-9b3f96b1f626)

We generate the build direct at the user's directory avoiding [this copy](https://github.com/event-catalog/eventcatalog/blob/main/bin/eventcatalog.ts#L114). What could be wrong with that???

CATALOG_DIR usually is `.eventcatalog-core`
PROJECT_DIR is the user's directory - usually `process.cwd()`

Reply:
Smart! I like that idea. I just tested it a bit locally and pushed it up.

Unrelated to this, but `npx eventcatalog build` fails if Im not in the project dir. It would be nice to be able to use catalog commands at any level of the project

Reply:
> Smart! I like that idea. I just tested it a bit locally and pushed it up.

Remove [this copy from build command](https://github.com/event-catalog/eventcatalog/blob/main/bin/eventcatalog.ts#L114) wasn't needed?


> Unrelated to this, but `npx eventcatalog build` fails if Im not in the project dir. It would be nice to be able to use catalog commands at any level of the project

What error you got it?